src/librustc_session/lint/builtin.rs

```diff
@@ -71,6 +71,12 @@ declare_lint! {
     "extern crates that are never used"
 }
 
+declare_lint! {
+    pub UNUSED_CRATE_DEPS,
```

Comment:
```suggestion
    pub UNUSED_EXTERN_OPTIONS,
```
https://github.com/rust-lang/rust/pull/69203 used `unused_extern_options` because that's what the lint does exactly (at least from rustc's point of view).
Crate dependencies may be introduced without the options (with `extern crate`) and unused `--extern` options may not even point to existing crates.

Reply:
I see what you mean, but referencing `--extern` is a bit obscure. `--extern` is not typically seen or used by end users, and they may not even be aware of them. Talking about "dependencies" is a bit higher level and more directly what the programmer is going to be using.

I'm not sure what you mean by `extern crate`. From my point of view, I see this lint as being responsible for highlighting a mismatch between the build system dependencies (as manifest by `--extern`) and what the source says - ie, does the source reference an external dependency? By this standard `extern crate` doesn't count because it's necessarily part of the sources - and we have `UNUSED_EXTERN_CRATES` for flagging those. Also I don't think it can actually add a new dependency can it? It's just a reference to one in the same manner as `use`, and so needs a corresponding `--extern`? (I guess std, core, etc excepted, but I think its reasonable for them to be exempt from this lint too.)

While `--extern` can refer to non-existent crates, and I think its useful to report that, it should never happen in practice unless the build system is broken, or you're manually invoking `rustc`.

Reply:
Ok, in that case `unused_crate_deps` should probably be renamed to `unused_crate_dependencies` to avoid jargon and match the `Cargo.toml` section where they are specified.